requests/auth.py

```diff
@@ -63,19 +64,26 @@ class HTTPDigestAuth(AuthBase):
     def __init__(self, username, password):
         self.username = username
         self.password = password
-        self.last_nonce = ''
-        self.nonce_count = 0
-        self.chal = {}
-        self.pos = None
-        self.num_401_calls = 1
+        # Keep state in per-thread local storage
+        self.tl = threading.local()
```

Comment:
@sigmavirus24 - thanks for the input, I was divided between the concise vs longer, more explicit name. I've updated the PR with commit 5a69137ac871c38f4597b574e93de07d9ff9655f following your suggestion.

Also: I threw away a single utilization of a local `num_401_calls` right before the `if 'digest' in ...` in `handle_401()`.